ui/raidboss/popup-text.ts

```diff
@@ -2,14 +2,86 @@ import { callOverlayHandler, addOverlayListener } from '../../resources/overlay_
 
 import AutoplayHelper from './autoplay_helper';
 import BrowserTTSEngine from './browser_tts_engine';
-import { addPlayerChangedOverrideListener } from '../../resources/player_override';
+import { addPlayerChangedOverrideListener, PlayerChangedDetail } from '../../resources/player_override';
 import PartyTracker from '../../resources/party';
 import Regexes from '../../resources/regexes';
 import Util from '../../resources/util';
 import ZoneId from '../../resources/zone_id';
+import {
+  LooseTrigger, OutputStrings, LocaleText, TriggerSet, TimelineFunc, LooseTriggerSet,
+  ResponseField, TriggerAutoConfig, MatchesAny, TriggerField, TriggerOutput,
+  TriggerFunc, OptionalUnlessVoid, Output, ResponseOutput, NetRegexTrigger, RegexTrigger,
+} from '../../types/trigger';
+import { UnreachableCode } from '../../resources/not_reached';
+import { Lang, NonEnLang } from '../../resources/languages';
+import { PerTriggerAutoConfig, PerTriggerOption, RaidbossOptions } from './raidboss_options';
+import { TimelineReplacement, TimelineLoader } from './timeline';
+import { RaidbossFileData } from './data/raidboss_manifest.txt';
+import { RaidbossData } from '../../types/data';
+import { Job, Role } from '../../types/job';
+import { EventResponses, LogEvent } from '../../types/event';
+
+export const isNetRegexTrigger =
+  (trigger?: LooseTrigger): trigger is NetRegexTrigger<RaidbossData> => {
```

Comment:
I don't know that this needs a new type, but it could be `Partial<NetRegexTrigger<RaidbossData>>` too.  Mostly I just wanted to future proof this in case it started getting used in other places where `id` is used.